lib/fog/openstack/models/image_v2/images.rb

```diff
@@ -18,9 +18,9 @@ def summary(options = {})
           end
 
           def find_by_id(id)
-            image_hash = service.get_image_by_id(id).body
-            Fog::Image::OpenStack::V2::Image.new(
-                image_hash.merge(:service => service))
+            result = all.find { |image| image.id == id }
```

Comment:
Fog core's `reload` function strangely requires a collection to be set on the model. Creating a model with `Image.new` would throw "undefined method 'get' on NilClass" when reload was called as [`collection` was nil]((https://github.com/fog/fog-core/blob/f6ef6c3d9458bc235be57100acdced264e85e52a/lib/fog/core/model.rb#L43). I'm not sure whether there's a better why to avoid the nil pointer exception short of using `all.find`.